### PR TITLE
Add context attribute to FencedFrameConfig

### PR DESCRIPTION
**Pull request by pythagoraskitty**
We propose adding a way for embedders to write context to the fenced frame config, which will then be available inside shared storage worklets in the fenced frame.

### CODE REVIEW

**Review by gtanzer — APPROVED** [review]

File: explainer/fenced_frame_config_context.md [hunk 1]
```diff
@@ -8,9 +8,9 @@ In a scenario where the input URLs for the fenced frame are required to be k-ano
...
+Subsequently, anything written through the [FencedFrameConfig](https://github.com/WICG/fenced-frame/blob/master/explainer/fenced_frame_config.md)'s `setSharedStorageContext()` method prior to a fenced frame's navigation to that config, can be read via `sharedStorage.context` from inside a worklet for the [Shared Storage API](https://github.com/WICG/shared-storage) created by the fenced frame or by any of its same-origin children. That is, `sharedStorage.context` can be invoked via the script for a operation previously registered through [`sharedStorage.worklet.addModule()`](https://github.com/WICG/shared-storage/tree/main#outside-the-worklet) and then called by either [`sharedStorage.run()`](https://github.com/WICG/shared-storage/tree/main#outside-the-worklet) or [`sharedStorage.selectURL()`](https://github.com/WICG/shared-storage/tree/main#outside-the-worklet).
```
Comment (shivanigithub) [inline explainer/fenced_frame_config_context.md:13]: Can we also add the same-origin restriction?

Comment (pythagoraskitty) [reply explainer/fenced_frame_config_context.md:13]: It notes the same-origin restriction in the previous sentences. Would you like me to repeat it?

Comment (pythagoraskitty) [reply explainer/fenced_frame_config_context.md:13]: "from inside a worklet for the [Shared Storage API](https://github.com/WICG/shared-storage) created by the fenced frame or by any of its same-origin children."

Comment (pythagoraskitty) [reply explainer/fenced_frame_config_context.md]: Just changed "children" -> "descendants" and removed `selectURL` per the chat discussion.

File: explainer/fenced_frame_config_context.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Subsequently, anything written through the [FencedFrameConfig](https://github.com/WICG/fenced-frame/blob/master/explainer/fenced_frame_config.md)'s `setSharedStorageContext()` method prior to a fenced frame's navigation to that config, can be read via `sharedStorage.context` from inside a worklet for the [Shared Storage API](https://github.com/WICG/shared-storage) created by the fenced frame or by any of its same-origin descendants. That is, `sharedStorage.context` can be invoked via the script for a operation previously registered through [`sharedStorage.worklet.addModule()`](https://github.com/WICG/shared-storage/tree/main#outside-the-worklet) and then called by either [`sharedStorage.run()`](https://github.com/WICG/shared-storage/tree/main#outside-the-worklet).
```
Comment (shivanigithub) [inline explainer/fenced_frame_config_context.md:13]: remove "either"

File: explainer/fenced_frame_config_context.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+## Example
+
+After the SSP's JavaScript invokes the [Turtledove/FLEDGE API](https://github.com/WICG/turtledove/blob/main/FLEDGE.md) and receives the resulting [FencedFrameConfig](https://github.com/WICG/fenced-frame/blob/master/explainer/fenced_frame_config.md) `fencedFrameConfig`, the SSP JS can write an event ID string via `fencedFrameConfig.setContext()`.
```
Comment (shivanigithub) [inline explainer/fenced_frame_config_context.md:17]: setContext  => setSharedStorageContext

File: explainer/fenced_frame_config_context.md (hunk 2 above)
Comment (pythagoraskitty) [reply explainer/fenced_frame_config_context.md]: Good catch, thank you!

File: explainer/fenced_frame_config_context.md (hunk 3 above)
Comment (pythagoraskitty) [reply explainer/fenced_frame_config_context.md]: Done

**Review by shivanigithub — APPROVED** [review]

**Pull request merged**